tremor-pipeline/src/query.rs

```diff
@@ -95,28 +95,41 @@ pub(crate) fn window_decl_to_impl<'script>(
         WindowKind::Tumbling => {
             let script = if d.script.is_some() { Some(d) } else { None };
             let ttl = d.params.get("eviction_period").and_then(Value::as_u64);
-
-            d.params
-                .get("interval")
+            let max_groups = d
+                .params
+                .get("max_groups")
                 .and_then(Value::as_u64)
-                .map_or_else(
-                    || {
-                        d.params.get("size").and_then(Value::as_u64).map_or_else(
-                            || {
-                                Err(Error::from(
-                            "Bad window configuration, either `size` or `interval` is required",
-                        ))
-                            },
-                            |size| {
-                                Ok(TumblingWindowOnNumber::from_stmt(size, ttl, script, stmt)
-                                    .into())
-                            },
-                        )
-                    },
-                    |interval| {
-                        Ok(TumblingWindowOnTime::from_stmt(interval, ttl, script, stmt).into())
-                    },
+                .unwrap_or(WindowImpl::DEFAULT_MAX_GROUPS);
+            let emit_empty_windows = d
+                .params
+                .get("emit_empty_windows")
+                .and_then(Value::as_bool)
+                .unwrap_or(WindowImpl::DEFAULT_EMIT_EMPTY_WINDOWS);
```

Comment:
we should emit a warning here if `emit_empty_windows` is set and `max_groups` is not set